src/core/geometry/qgsellipse.h

```diff
@@ -0,0 +1,219 @@
+     * The center point can have z and m values which are the result from the midpoint operation between \a pt1 and \a pt2.
+     * Axes are calculated from the 2D distance with the third point \a pt3.
+     * The azimuth is the angle between \a pt1 and \a pt2.
+     * @param pt1 First focus.
+     * @param pt2 Second focus.
+     * @param pt3 A point to calculate the axes.
+     */
+    static QgsEllipse fromFoci( const QgsPointV2 &pt1, const QgsPointV2 &pt2, const QgsPointV2 &pt3 );
+
+    /**
+     * Constructs an ellipse by an extent (aka bounding box / QgsRectangle).
+     * The center point can have z and m values which are the result from the midpoint operation between \a pt1 and \a pt2.
+     * Axes are calculated from the 2D distance between \a pt1 and \a pt2.
+     * The azimuth always takes the default value.
+     * @param pt1 First corner.
+     * @param pt2 Second corner.
+     */
+    static QgsEllipse byExtent( const QgsPointV2 &pt1, const QgsPointV2 &pt2 );
+
+    /**
+     * Constructs an ellipse by a center point and a another point.
+     * The center point keep z and m values from \a ptc.
+     * Axes are calculated from the 2D distance between \a ptc and \a pt1.
+     * The azimuth always takes the default value.
+     * @param ptc Center point.
+     * @param pt1 First point.
+     */
+    static QgsEllipse byCenterPoint( const QgsPointV2 &ptc, const QgsPointV2 &pt1 );
+
+    /**
+     * Constructs an ellipse by a central point and two other points.
+     * The center point keep z and m values from \a ptc.
+     * Axes are calculated from the 2D distance between \a ptc and \a pt1 and \a pt2.
+     * The azimuth is the angle between \a ptc and \a pt1.
+     * @param ptc Center point.
+     * @param pt1 First point.
+     * @param pt2 Second point.
+     */
+    static QgsEllipse byCenter2Points( const QgsPointV2 &ptc, const QgsPointV2 &pt1, const QgsPointV2 &pt2 );
+
+    virtual bool operator ==( const QgsEllipse &elp ) const;
+    virtual bool operator !=( const QgsEllipse &elp ) const;
+
+    //! An ellipse is empty if axes are equal to 0
+    virtual bool isEmpty() const;
+
+    /** Returns the center point.
+     * @see setCenter()
+     * @see rcenter()
+     */
+    QgsPointV2 center() const {return mCenter; }
+
+    /** Returns the semi-major axis.
+     * @see setSemiMajorAxis()
+     */
+    double semiMajorAxis() const {return mSemiMajorAxis; }
+
+    /** Returns the semi-minor axis.
+     * @see setSemiMinorAxis()
+     */
+    double semiMinorAxis() const {return mSemiMinorAxis; }
+
+    /** Returns the azimuth.
+     * @see setAzimuth()
+     */
+    double azimuth() const {return mAzimuth; }
+
+    /** Returns a reference to the center point of this ellipse.
+     * Using a reference makes it possible to directly manipulate center in place.
+     * @see center()
+     * @see setCenter()
+     * @note not available in Python bindings
+     */
+    QgsPointV2 &rcenter() { return mCenter; }
```

Comment:
`SIP_SKIP` should do the job
like [here](https://github.com/qgis/QGIS/blob/33b6c41dd2b516e5cc141b1550bcd6d6b9679955/src/core/qgsmaplayer.h#L141)